tests/ui/recursion/recursive-enum-box.rs

```diff
@@ -0,0 +1,20 @@
+//@ run-pass
+// A smoke test for recursive enum structures using Box<T>.
+// This test constructs a linked list-like structure to exercise memory allocation and ownership.
+// Originally introduced in 2010, this is one of Rust’s earliest test cases.
+
+enum List {
+    #[allow(dead_code)]
```

Comment:
Nit: I would just hoist this as

```rs
#![allow(dead_code)]
```

This test isn't about exercising `dead_code`-the-lint really